### PR TITLE
Use git rev-parse --show-toplevel for the root dir.

### PR DESCRIPTION
**Pull request by mccheah**
Makes it so that the test scripts do not necessarily have to be run from the top level of the repository.

### CODE REVIEW

File: e2e/runner.sh [hunk 1]
```diff
@@ -79,44 +74,43 @@ echo "Running tests on cluster $MASTER against $REPO."
...
 else
   # -m option for minikube.
-  cd dist && ./sbin/build-push-docker-images.sh -m -r $IMAGE_REPO -t $tag build
+  cd dist && ./sbin/build-push-docker-images.sh -m -r $IMAGE_REPO -t $TAG build
 fi
 
-cd $root/integration-test
-$root/spark/build/mvn clean -Ddownload.plugin.skip=true integration-test \
-          -Dspark-distro-tgz=$root/spark/*.tgz \
+$TEST_ROOT/integration-test/build/mvn clean -Ddownload.plugin.skip=true integration-test \
```
Comment (foxish) [inline e2e/runner.sh]: Should be  `$TEST_ROOT/spark/build/mvn`

Comment (foxish) [reply e2e/runner.sh]: or `$SPARK_REPO_ROOT/build/mvn` perhaps

Comment (mccheah) [reply e2e/runner.sh]: Ah that's confusing because we're using the Maven bundled with Spark. We should just provide our own, no?

Comment (foxish) [reply e2e/runner.sh]: I tried starting with that - but the two mavens I tried were incompatible and caused the spark-build to stall. This works for now and I like that we'll be using the same maven as the spark build, so, moving it in-repo will be easier as well.

**Review by foxish — APPROVED** [review]
Thanks! LGTM

**Pull request merged**